nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1231,13 +1232,14 @@ async fn ip_pool_ranges_get(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// TODO shouldn't this just be a POST on /ip-pools/{pool_name}/ranges?
```

Comment:
Dave, Robert, and I talked a bit about this. It seemed distinct enough from the standard REST API to warrant a POST on specific endpoints, similar to instance reboot or disk attach. Part of it was that the ranges don't have names like the rest of the resources, they're identified by a UUID and, currently by the start/end address. POST-ing to `../ranges` to create a new one seems OK. DELETE would require either that clients use the UUID, or accepting a body, which was why we opted for POST.

Another thing pushing us this way was that potentially, adding or removing a range is not a "simple" add / remove operation. Addition could be expanded to include _merging_ neighboring ranges. Deletion could include splitting ranges. The motivation for the latter was Robert pointing out that sometimes one wants to claw back a specific IP from within a range, and requiring that to be a DELETE followed by two POSTs seemed problematic.

I'm not dogmatic about this sort of thing though, whatever makes the most sense. @davepacheco  Did I capture that well enough.

Reply:
That makes sense. Thanks. How about this:

```diff
 ip_pool_create                           /ip-pools
 ip_pool_delete                           /ip-pools/{pool_name}
 ip_pool_list                             /ip-pools
-ip_pool_range_create                     /ip-pools/{pool_name}/ranges/add
-ip_pool_range_delete                     /ip-pools/{pool_name}/ranges/delete
+ip_pool_range_add                        /ip-pools/{pool_name}/ranges/add
 ip_pool_range_list                       /ip-pools/{pool_name}/ranges
+ip_pool_range_remove                     /ip-pools/{pool_name}/ranges/remove
 ip_pool_update                           /ip-pools/{pool_name}
 ip_pool_view                             /ip-pools/{pool_name}
```

In short: I changed `delete` to `remove` to avoid the confusion

Reply:
Yeah, that sounds like a good approach, thanks!